src/Faithlife.Data/DbConnectorCommand.cs

```diff
@@ -302,7 +302,7 @@ private IReadOnlyList<T> DoQuery<T>(Func<IDataRecord, T> read)
 			return list;
 		}
 
-		private T DoQueryFirst<T>(Func<IDataRecord, T> read, bool single, bool orDefault)
+		private T DoQueryFirst<T>(Func<IDataRecord, T>? read, bool single, bool orDefault)
```

Comment:
Need to add `[return: MaybeNull]`?

Reply:
I toyed with that but it wasn't clear what good it was doing. I'd rather not add it until I understand why it is useful.

Reply:
You're coercing `default(T)` with `!` in this method, so it's pretty clear the method can return `null`. The `MaybeNull` attribute is designed for this exact use case.

I'd rather you not add nullable annotations until you understand why the attributes are needed.

Reply:
I understand the fundamental principle behind the attribute, but documentation is sparse, the attribute wasn't having the effect I expected (see below), and I can't use it when building .NET Standard 2.0, so I let it go. Happy for any assistance you can give.

Reply:
da8c6694249ce9936550a595119d4c2e93407a99